core/src/main/java/hudson/model/EnvironmentContributingAction.java

```diff
@@ -45,12 +50,39 @@
  */
 public interface EnvironmentContributingAction extends Action {
     /**
+     * Called by {@link Run} or {@link AbstractBuild} to allow plugins to contribute environment variables.
+     *
+     * @param run
+     *      The calling build. Never null.
```

Comment:
Instead of "Never null" it's better to add the `@Nonnull` annotation to the method argument